src/arch/time.c

```diff
@@ -17,12 +16,37 @@ _v; \
 })
 
 #define CMOS_READ(addr) ({ \
-    outb_p(0x80|addr,0x70); \
+    outb_p(addr,0x70); \
     inb_p(0x71); \
 })
 
+uint8_t hour, minute, second, registerB, day, month, year;
 void show_time() {
-    kprintf("%03i/%03i/%03i %03i:%03i:%03i\n",
-            CMOS_READ(8), CMOS_READ(7), CMOS_READ(9),
-            CMOS_READ(4), CMOS_READ(2), CMOS_READ(0));
+	// referenced from : https://wiki.osdev.org/CMOS#RTC_Update_In_Progress
+	// reading values from associated CMOS register number containing respective data
+	hour = CMOS_READ(0x04);
+	minute = CMOS_READ(0x02);
+	second = CMOS_READ(0x00);
+
+	day = CMOS_READ(0x07);
+	month = CMOS_READ(0x08);
+	year = CMOS_READ(0x09);
+	
+
+	// Converting BCD to binary values
+	second = (second & 0x0F) + ((second / 16) * 10);
+	minute = (minute & 0x0F) + ((minute / 16) * 10);
+	hour = ( (hour & 0x0F) + (((hour & 0x70) / 16) * 10) ) | (hour & 0x80);
+
+	day = (day & 0x0F) + ((day / 16) * 10);
+	month = (month & 0x0F) + ((month / 16) * 10);
+	year = (year & 0x0F) + ((year / 16) * 10);
+	
+	// Converting 12 hour clock to 24 hour clock 
+	registerB = CMOS_READ(0x0B);
```

Comment:
just inline this into the next line, ie:
if (!(CMOS_READ(0x08) & 0x02) && (hour & 0x80))

Reply:
Didn't see that it was totally unnecessary declaring that "registerB" separately.